Production_scripts/B01_acute_draft.R

```diff
@@ -0,0 +1,332 @@
+    lca = `Geo Council Area Code`,
+    DataZone = `Geo Data Zone 2011`,
+    postcode = `Geo Postcode [C]`,
+    HSCP = `Geo HSCP of Residence Code - current`,
+    conc = `Lead Consultant/HCP Code`,
+    admloc = `Location Admitted Trans From Code`,
+    dischloc = `Location Discharged Trans To Code`,
+    mpat = `Management of Patient Code`,
+    hbrescode = `NHS Board of Residence Code - current`,
+    nhshosp = `NHS Hospital Flag (01)`,
+    yearstay = `Occupied Bed Days (01)`,
+    oldtadm = `Old SMR1 Type of Admission Code`,
+    op1a = `Operation 1A Code (4 char)`,
+    op1b = `Operation 1B Code (4 char)`,
+    op2a = `Operation 2A Code (4 char)`,
+    op2b = `Operation 2B Code (4 char)`,
+    op3a = `Operation 3A Code (4 char)`,
+    op3b = `Operation 3B Code (4 char)`,
+    op4a = `Operation 4A Code (4 char)`,
+    op4b = `Operation 4B Code (4 char)`,
+    gender = `Pat Gender Code`,
+    chi = `Pat UPI`,
+    cat = `Patient Category Code`,
+    gpprac = `Practice Location Code`,
+    hbpraccode = `Practice NHS Board Code - current`,
+    selfharm_adm = `Self Harm Related Admission (01)`,
+    sigfac = `Significant Facility Code`,
+    spec = `Specialty Classificat. 1/4/97 Code`,
+    submis_adm = `Substance Misuse Related Admission (01)`,
+    cost_total_net = `Total Net Costs (01)`,
+    location = `Treatment Location Code`,
+    hbtreatcode = `Treatment NHS Board Code - current`,
+    uri = `Unique Record Identifier`,
+    record_keydate1 = `Date of Admission(01)`,
+    record_keydate2 = `Date of Discharge(01)`,
+    dateop1 = `Date of Operation 1 (01)`,
+    dateop2 = `Date of Operation 2 (01)`,
+    dateop3 = `Date of Operation 3 (01)`,
+    dateop4 = `Date of Operation 4 (01)`,
+    dob = `Pat Date Of Birth [C]`,
+    ipdc = `Inpatient Day Case Identifier Code`,
+    cij_ipdc = `CIJ Inpatient Day Case Identifier Code (01)`,
+    lineno = `Line Number (01)`,
+    GLS_record = `GLS Record`
+  )
+
+
+# Data Cleaning ---------------------------------------
+
+acute_clean <- acute_file %>%
+  # Set year variable
+  mutate(
+    year = year,
+    # Set recid as 01B and flag GLS records
+    recid = if_else(GLS_record == "Y", "GLS", "01B"),
+    # Set IDPC marker for the episode
+    ipdc = case_when(
+      ipdc == "IP" ~ "I",
+      ipdc == "DC" ~ "D"
+    ),
+    # Set IDPC marker for the cij
+    cij_ipdc = case_when(
+      cij_ipdc == "IP" ~ "I",
+      cij_ipdc == "DC" ~ "D"
+    )
+  ) %>%
+  # Recode GP practice into 5 digit number
+  # We assume that if it starts with a letter it's an English practice and so recode to 99995.
+  convert_eng_gpprac_to_dummy(gpprac) %>%
+  # Calculate the total length of stay (for the entire episode, not just within the financial year).
+  mutate(
+    stay = difftime(record_keydate2, record_keydate1, units = "days"),
+    # create and populate SMRType
+    SMRType = case_when(
+      recid == "01B" & lineno != 330 ~ if_else(ipdc == "I", "Acute-IP", "Acute-DC")
```

Comment:
```suggestion
      recid == "01B" & lineno != 330 ~ if_else(ipdc == "I", "Acute-IP", "Acute-DC"),
```

missing comma in here I think? code was breaking for me